ocaml/xapi/repository_helpers.ml

```diff
@@ -634,6 +634,11 @@ let eval_guidances ~updates_info ~updates ~kind ~livepatches ~failed_livepatches
   in
   let upd_ids_of_livepatches = extract_upd_ids livepatches in
   let upd_ids_of_failed_livepatches = extract_upd_ids failed_livepatches in
+  let initial_gs =
+    List.fold_left
```

Comment:
How about creating a set from a list first and joining the sets? Using `GuidanceSet.of_list` to create the set.

Reply:
@lindig , thanks for the suggestion~ But to create a set with `GuidanceSet.of_list`, we need to transfer from "update_guidance" to "Guidance" (with `Guidance.of_update_guidance`) first.